### ISSUE TITLE
Broken yield parsing in python 2.7

### ISSUE THREAD

**Issue by kylec1** [JukkaL/mypy#751]
Code:
  
     def wrapped(*args, **kwargs):
          with new_scope(g, r):
                ret = yield f(*args, **kwargs)

results in ` error: Parse error before "f"`.

**Comment by o11c** [comment]
Duplicate of #498 which I am already working on.

**Comment by JukkaL** [comment]
We are are currently blocked on this, so I'll implement a quick hack the lets mypy process code such as the example above, but doesn't actually do useful type checking.
